dev/tests/integration/testsuite/Magento/Catalog/Model/Product/Option/OptionsSavingTest.php

```diff
@@ -0,0 +1,336 @@
+use Magento\TestFramework\Helper\Bootstrap;
+use PHPUnit\Framework\TestCase;
+
+class OptionsSavingTest extends TestCase
+{
+    /**
+     * @var ObjectManagerInterface
+     */
+    private $objectManager;
+
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $this->objectManager = Bootstrap::getObjectManager();
+        $this->productRepository = $this->objectManager->get(ProductRepositoryInterface::class);
+    }
+
+    /**
+     * @magentoDataFixture Magento/Catalog/_files/my_product_with_options.php
+     * @magentoDbIsolation enabled
+     * @return void
+     */
+    public function testUpdateField(): void
+    {
+        $product = $this->productRepository->get('simple');
+        $options = $product->getOptions();
+        $optionsQty = count($options);
+        $found = null;
+        foreach ($options as $option) {
+            if ($option->getTitle() === 'test_option_code_1') {
+                $found = $option;
+            }
+        }
+        $this->assertNotNull($found);
+        $found->setTitle('Cahnged');
+        $product->setOptions($options);
+        $this->productRepository->save($product);
+        $product = $this->productRepository->get('simple', false, null, true);
+        $options = $product->getOptions();
+        $found = null;
+        foreach ($options as $option) {
+            if ($option->getTitle() === 'Cahnged') {
+                $found = $option;
+            }
+        }
+        $this->assertNotNull($found);
+        $this->assertCount($optionsQty, $options);
+    }
+
+    /**
+     * @magentoDataFixture Magento/Catalog/_files/my_product_with_options.php
+     * @magentoDbIsolation enabled
+     * @return void
+     */
+    public function testUpdateArea(): void
+    {
+        $product = $this->productRepository->get('simple');
+        $options = $product->getOptions();
+        $optionsQty = count($options);
+        $found = null;
+        foreach ($options as $option) {
+            if ($option->getTitle() === 'area option') {
+                $found = $option;
+            }
+        }
+        $this->assertNotNull($found);
+        $found->setTitle('Cahnged');
+        $product->setOptions($options);
+        $this->productRepository->save($product);
+        $product = $this->productRepository->get('simple', false, null, true);
+        $options = $product->getOptions();
+        $found = null;
+        foreach ($options as $option) {
+            if ($option->getTitle() === 'Cahnged') {
+                $found = $option;
+            }
+        }
+        $this->assertNotNull($found);
+        $this->assertCount($optionsQty, $options);
+    }
+
+    /**
+     * @magentoDataFixture Magento/Catalog/_files/my_product_with_options.php
+     * @magentoDbIsolation enabled
+     * @return void
+     */
+    public function testUpdateDropDown(): void
```

Comment:
You should update option value (not option) and check it.
Also use ProductCustomOptionRepositoryInterface for get options.